tools/refguide_check.py

```diff
@@ -150,43 +151,155 @@ def report(all_dict, funcnames, deprecated, module_name):
-          'assert_equal': np.testing.assert_equal}
+          'assert_equal': np.testing.assert_equal,
+          # recognize numpy repr's
+          'array': np.array,
+          'int64': np.int64,
+          'uint64': np.uint64,
+          'int8': np.int8,
+          'int32': np.int32,
+          'float64': np.float64,
+          'dtype': np.dtype,
+          'nan': np.nan, 'NaN': np.nan,
+          'inf': np.inf, 'Inf': np.inf,}
+
+    # if MPL is available, use display-less backend
+    try:
+        import matplotlib
+        matplotlib.use('Agg')
+        import matplotlib.pyplot as plt
+        have_MPL = True
+    except ImportError:
+        have_MPL = False
+
+
+    class DTRunner(doctest.DocTestRunner):
+        stopwords = {'plt.', '.hist', '.show', '.ylim', '.subplot(',
+                     'set_title', 'imshow', 'plt.show', 'ax.axis', 'plt.plot(',
+                     '.title', '.ylabel', '.xlabel', 'set_ylim', 'set_xlim'}
+        rndm_markers = {'# random', '# Random', '#random', '#Random'}
+
+        def report_failure(self, out, test, example, got):
+            if (any(word in example.source for word in self.stopwords) or
+                any(word in example.want for word in self.rndm_markers)):
+                # do not complain if output does not match
+                pass
+            else:
+                return doctest.DocTestRunner.report_failure(self, out, test,
+                                                            example, got)
+
+    class Checker(doctest.OutputChecker):
+        obj_pattern = re.compile('at 0x[0-9a-fA-F]+>')
+        vanilla = doctest.OutputChecker()
+
+        def __init__(self, parse_namedtuples=True, atol=1e-8, rtol=1e-2):
+            self.parse_namedtuples = parse_namedtuples
+            self.atol, self.rtol = atol, rtol
+
+        def check_output(self, want, got, optionflags):
+
+            # cut it short if they are equal
+            if want == got:
+                return True
+
+            # skip function/object addresses
+            if self.obj_pattern.search(got):
+                return True
+
+            # ignore comments (e.g. signal.freqresp)
+            if want.lstrip().startswith("#"):
+                return True
+
+            # try the standard doctest
+            try:
+                if self.vanilla.check_output(want, got, optionflags):
+                    return True
+            except Exception:
+                pass
+
+            # OK then, convert strings to objects
+            try:
+                a_want = eval(want, ns)
+                a_got = eval(got, ns)
+            except:
+                if not self.parse_namedtuples:
+                    return False
+                # suppose that "want"  is a tuple, and "got" is smth like
+                # MoodResult(statistic=10, pvalue=0.1).
+                # Then convert the latter to the tuple (10, 0.1), 
+                # and then compare the tuples.
+                try:
```

Comment:
I thought I'd draw attention to this code path, which goes to some lengths to compare, e.g. `(10, 0.1)` and `MoodResult(statistic=10, pvalue=0.1)` by values rather than literally. Without this mess, checking `stats` fails all around because of namedtuples. My logic here is that specific names of the namedtuples are not a part of the API, they are just an artifact of that we cannot have a nameless object with named attributes. Hence we should only check the numeric values.